lib/src/collect.dart

```diff
@@ -65,19 +66,37 @@ Future<Map<String, dynamic>> collect(
 }
 
 Future<Map<String, dynamic>> _getAllCoverage(
-    VmService service, bool includeDart) async {
+    VmService service, bool includeDart, Set<String> scopedOutput) async {
+  scopedOutput ??= Set<String>();
   final vm = await service.getVM();
   final allCoverage = <Map<String, dynamic>>[];
 
   for (var isolateRef in vm.isolates) {
-    final SourceReport report = await service.getSourceReport(
-      isolateRef.id,
-      <String>[SourceReportKind.kCoverage],
-      forceCompile: true,
-    );
-    final coverage =
-        await _getCoverageJson(service, isolateRef, report, includeDart);
-    allCoverage.addAll(coverage);
+    if (scopedOutput.isNotEmpty) {
+      final scripts = await service.getScripts(isolateRef.id);
+      for (var script in scripts.scripts) {
+        final uri = Uri.parse(script.uri);
+        if (uri.scheme != 'package') continue;
+        final scope = uri.path.split('/').first;
+        // Skip scripts which should not be included in the report.
+        if (!scopedOutput.contains(scope)) continue;
+        final scriptReport = await service.getSourceReport(
```

Comment:
This may not matter too much, but you could try and run a few concurrently by building up a list of futures and then awaiting the result. Something like.

var pending = <Future<void>>[];
...
pending.add(service.getSourceReport(...).then((scriptReport) {
  return _getCoverageJson(....);
}).then(allCoverage.addAll));
...
await Future.wait(pending);

Reply:
Doing some internal testing this suggestion surprisingly seemed to slow things down.